server/src/metrics/metric-result.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+export abstract class MetricResult {
+  /**
+   * Prints a summary of the metric results
+   */
+  abstract logSummary(): void;
+
+  /**
+   * Prints a more verbose set of data used for the Metric results
+   */
+  abstract logRawData(): void;
```

Comment:
These should probably return strings? @aomarks wdyt?

Reply:
Maybe they should take a `NodeJS.WritableStream`. Then the CLI can pass e.g. `process.stdout` `fs.createWriteStream('whatever.log')`.

https://nodejs.org/api/stream.html#stream_writable_streams

Reply:
strings seem simpler for now IMO.

Reply:
+1 string. I doubt there will be a scenario that a writable stream would be beneficial